### ISSUE TITLE
【Markdown】It can't skip to the second one when there are two same heading.

### ISSUE THREAD

**Issue by imhuay** [Microsoft/vscode#59711]
- VSCode Version: 1.27.2
- OS Version:

Steps to Reproduce:

```
<!-- TOC -->

- [Test](#test)
- [Test](#test-1)

<!-- /TOC -->

## Test

## Test
```
- The second Test can't be skipped to.

Does this issue occur when all extensions are disabled?: Yes

Another, when I use `code --disable-extensions` to start, it can't recover to click the reload button. It have to restart the vscode.

**Comment by denosawr** [comment]
I assume you're referring to the Markdown Preview display?

**Comment by denosawr** [comment]
I've done a little work looking into this, and it seems that there are two primary issues. First, VSCode's slugifier function doesn't add a suffix to each header's ID. Second, the underlying module VSC uses [`markdown-it-named-headers`](https://github.com/leff/markdown-it-named-headers) doesn't actually... seem to work... in regards to its functionality to provide a `used_headers` variable to the slugifier function - see https://github.com/leff/markdown-it-named-headers/issues/6.

**Comment by mjbvz** [comment]
Fixed for editor with 7d95e3e

Support for this in the preview is #47537 which I'll also look into this month. Thanks for the additional analysis @denosawr
